Models/Consumable.cs

```diff
@@ -1,79 +1,80 @@
 		public ConsumableType? Type;
 		public Color? CustomTextColor;
-		public Func<bool> HasBeenConsumed;
+		public Func<int> CurrentCount; // How many of the item have been consumed; for bools (aka single use consumables), this is either 0 or 1
+		public Func<int> TotalCount; // Total number of items that can be consumed; for bools, this is 1
 
 		public bool UsingMissingTexture;
 
-		public Consumable(string displayText, string texturePath, (float X, float Y) assetDimensions, string hoverText, ConsumableType type, Func<bool> hasBeenConsumed) {
-			DisplayText = displayText;
-			//TexturePath = texturePath;
-			//AssetDimensions = assetDimensions;
-			HoverText = hoverText;
-			Type = type;
-			CustomTextColor = null;
-			HasBeenConsumed = hasBeenConsumed;
+		#region Init
+		public Consumable(ModItem modItem, object CategoryOrCustomColor, Func<int> currentCount, Func<int> totalCount) {
+			DisplayText = modItem.DisplayName.ToString();
+			HoverText = modItem.Tooltip.ToString();
+			CurrentCount = currentCount;
+			TotalCount = totalCount;
+			SetTextureAndDimensions(modItem.Texture);
 
-			SetTextureAndDimensions(texturePath: texturePath, assetDimensions: assetDimensions);
+			if (CategoryOrCustomColor is Color color) CustomTextColor = color;
+			else if (CategoryOrCustomColor is string categoryName) Type = GetType(categoryName);
 		}
 
-		public Consumable(string displayText, string texturePath, (float X, float Y) assetDimensions, string hoverText, Color customHoverColor, Func<bool> hasBeenConsumed) {
-			DisplayText = displayText;
-			//TexturePath = texturePath;
-			//AssetDimensions = assetDimensions;
-			HoverText = hoverText;
-			Type = null;
-			CustomTextColor = customHoverColor;
-			HasBeenConsumed = hasBeenConsumed;
-
-			SetTextureAndDimensions(texturePath: texturePath, assetDimensions: assetDimensions);
+		public Consumable(int vanillaItemId, ConsumableType type, Func<int> currentCount, Func<int> totalCount) {
+			DisplayText = Lang.GetItemName(vanillaItemId).Value;
+			HoverText = Language.GetText($"ItemTooltip.{ItemID.Search.GetName(vanillaItemId)}").Value;
+			CurrentCount = currentCount;
+			TotalCount = totalCount;
+			SetTextureAndDimensions(vanillaItemId);
+			Type = type;
 		}
+		#endregion
 
-		public Consumable(VanillaConsumable vc) {
-			DisplayText = vc.DisplayText;
-			//TexturePath = vc.TexturePath;
-			//AssetDimensions = vc.AssetDimensions;
-			HoverText = vc.HoverText;
-			Type = vc.Type;
-			CustomTextColor = null;
-			HasBeenConsumed = vc.HasBeenConsumed;
-
-			SetTextureAndDimensions(texturePath: vc.TexturePath, assetDimensions: vc.AssetDimensions);
-		}
+		#region Public helpers
+		public bool HasBeenConsumed => CurrentCount() == TotalCount();
+		public bool IsMultiUse => TotalCount() > 1;
+		#endregion
 
-		private void SetTextureAndDimensions(string texturePath, (float X, float Y) assetDimensions) {
+		#region Private helpers
+		private void SetTextureAndDimensions(string texturePath) {
```

Comment:
Can be renamed to just `SetTexture`. Same with the other method of the same name